src/internal/connector/onedrive/restore.go

```diff
@@ -42,11 +47,29 @@ func RestoreCollections(
 		restoreErrors = support.WrapAndAppend(id, err, restoreErrors)
 	}
 
+	// Reorder collections so that the parents directories are created
+	// before the child directories
+	sort.Slice(dcs, func(i, j int) bool {
+		return dcs[i].FullPath().String() < dcs[j].FullPath().String()
+	})
+
+	parentPermissions := map[string][]UserPermission{}
+
 	// Iterate through the data collections and restore the contents of each
 	for _, dc := range dcs {
-		temp, canceled := RestoreCollection(ctx, service, dc, OneDriveSource, dest.ContainerName, deets, errUpdater)
+		parentPerms, ok := parentPermissions[dc.FullPath().String()]
+		if !ok {
+			errUpdater(dc.FullPath().String(), fmt.Errorf("unable to find parent permissions"))
```

Comment:
this may need relaxed if we're only storing permissions when they're not the default

Reply:
As of now I am storing the metadata even if it is just the defaults, but it does not really make much sense to store it if there is no change. I'll probably have that in a follow up PR as we might end up changing a few things related to how we update the metrics success.

Reply:
We might also have to think about if we keep the metadata file in the details if we are going this route. If we are not always storing the metadata, I think we will need to know when to ask for the meta file from kopia.

Reply:
ah, perhaps I misinterpreted some of the other code for fetching permissions or something then. I think what we're kind of running up against here is how we ask kopia for things that we may not know the name of (i.e. "select all files with a `.meta` suffix`). This is something that we may (eventually) need to expand kopia to handle since it seems like email attachments will land in a similar situation (ignoring the ordering issue of things for the moment)

For now, it's fine to always store permissions, even if they're the default. We can come back and optimize later if we find a need for it